main.go

```diff
@@ -0,0 +1,382 @@
+		log.Info("No Thanos records to register, canceling run")
+		return nil
+	}
+	route53Targets := addPortToTargets(thanosRecords)
+
+	// This part can be used for local testing
+	// kubeconfig := filepath.Join(
+	// 	os.Getenv("HOME"), ".kube", "config",
+	// )
+
+	// config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	// if err != nil {
+	// 	return err
+	// }
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return errors.Wrap(err, "Unable to set k8s access config")
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "Unable to create k8s clientset")
+	}
+
+	configMapTargets, err := getConfigMapTargets(thanosConfigMapName, thanosNamespace, clientset)
+	if err != nil && !errors.IsNotFound(err) {
+		return errors.Wrap(err, "Unable to get existing configmap targets")
+	}
+
+	configMapUpToDate := checkIfConfigMapUpToDate(configMapTargets, route53Targets)
+
+	if len(configMapTargets) < 1 || !configMapUpToDate {
+		log.Info("The Thanos configmap targets are not up to date with existing Route53 Thanos targets")
+		err = createOrUpdateTargetConfigMap(route53Targets, thanosConfigMapName, thanosNamespace, clientset)
+		if err != nil {
+			return errors.Wrap(err, "Unable to create or update the Thanos Query configmap")
+		}
+
+		err = rotateQueryPods(thanosNamespace, thanosDeploymentName, clientset)
+		if err != nil {
+			return errors.Wrap(err, "Unable to rotate Thanos Query pods")
+		}
+
+		log.Info("Successfully rotated all Thanos Query pods")
+	} else {
+		log.Info("The Thanos configmap targets are up to date with existing Route53 Thanos targets")
+	}
+	return nil
+}
+
+// checkIfConfigMapUpToDate is used to check if the current configmap targets are up to date with the deploye Route53 targets.
+func checkIfConfigMapUpToDate(configMapTargets, route53Targets []string) bool {
+	if len(route53Targets) != len(configMapTargets) {
+		return false
+	}
+
+	for i, v := range route53Targets {
+		if v != configMapTargets[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// getConfigMapTargets is used to get the existing Thanos Query targets defined in the configmap
+func getConfigMapTargets(thanosConfigMapName, thanosNamespace string, clientset *kubernetes.Clientset) ([]string, error) {
+	ctx := context.TODO()
+	configMap, err := clientset.CoreV1().ConfigMaps(thanosNamespace).Get(ctx, thanosConfigMapName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, errors.Wrap(err, "Unable to get the Thanos Query configmap")
+	}
+	data := (configMap.Data["servicediscovery.yml"])
+	configStructure := config{}
+	log.Info("Decoding configmap data into structure")
+	configDecoded, err := decodeConfig(data, configStructure)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to decode into struct")
+	}
+	log.Info("Successfully decoded configmap data into structure")
+	if len(configDecoded) > 0 {
+		return configDecoded[0].Targets, nil
+	}
+	return []string{}, nil
```

Comment:
if we return `nil` here that means that the `getConfigMapTargets` was successful, right?